pkg/capabilities/consensus/ocr3/reduce/reduce_aggregator.go

```diff
@@ -0,0 +1,392 @@
+package reduce
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"math/big"
+	"reflect"
+	"sort"
+	"time"
+
+	"github.com/shopspring/decimal"
+	"google.golang.org/protobuf/proto"
+
+	ocrcommon "github.com/smartcontractkit/libocr/commontypes"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/consensus/ocr3/types"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+	"github.com/smartcontractkit/chainlink-common/pkg/values/pb"
+)
+
+type AggregatorConfig struct {
+	// Static data that is added into every report
+	AdditionalData map[string]any `mapstructure:"additionalData"`
+	// Configuration on how to aggregate one or more data points
+	Fields []AggregationField `mapstructure:"fields"`
+	// The top level field name that report data is put into
+	OutputFieldName string `mapstructure:"outputFieldName" json:"outputFieldName" default:"Reports"` // TODO: is this actually needed?
+	// The structure surrounding the report data that is put on to "OutputFieldName"
+	ReportFormat string `mapstructure:"reportFormat" json:"reportFormat" default:"map" jsonschema:"enum=map,enum=array"`
```

Comment:
Why is this needed? Feels like extra complexity, but I may well be missing the point :/

Reply:
It _is_ extra complexity that I don't love having in -- but the purpose is:

The Data Feeds consumer expects that EncodableOutcome looks like:
```
{
"Reports": [
      {
         "FeedID": []byte,
         "Price": *big.Int,
         "Timestamp": int64
      },
...
}
```

Notice that it's an array of reports, so that it can handle multiple mercury reports coming through.

The EVM Encoder gets the config:
![image](https://github.com/user-attachments/assets/410eae2c-cce0-4fcf-9d6e-41282b5bc661)

So this Aggregator config is to tell it to wrap in an array, even if there's only one report.

Reply:
Ah I see, sorry I missed that initially. Makes sense 👍